modules/charon-core/src/test/java/org/wso2/charon3/core/utils/PatchOperationUtilTest.java

```diff
@@ -0,0 +1,448 @@
+        AttributeSchema subAttributeSchema2 =
+                SCIMAttributeSchema.createSCIMAttributeSchema(
+                        "urn:ietf:params:scim:schemas:extension:enterprise:2.0:User:department",
+                        "department", STRING, false, "", false, false,
+                        READ_WRITE, DEFAULT, NONE, null, null, null);
+        AttributeSchema subAttributeSchema3 =
+                SCIMAttributeSchema.createSCIMAttributeSchema(
+                        "urn:ietf:params:scim:schemas:core:2.0:User:emails.value",
+                        "value", STRING, false, "", false, false,
+                        READ_WRITE, DEFAULT, NONE, null, null, null);
+        ArrayList<AttributeSchema> subAttributeSchemaList = new ArrayList<>();
+        subAttributeSchemaList.add(subAttributeSchema1);
+        subAttributeSchemaList.add(subAttributeSchema2);
+        ArrayList<AttributeSchema> subAttributeSchemaList1 = new ArrayList<>();
+        subAttributeSchemaList1.add(subAttributeSchema3);
+        AttributeSchema attributeSchema1 =
+                SCIMAttributeSchema.createSCIMAttributeSchema(
+                        "urn:ietf:params:scim:schemas:extension:enterprise:2.0:User",
+                        "urn:ietf:params:scim:schemas:extension:enterprise:2.0:User", COMPLEX,
+                        false, "", false, false,
+                        READ_WRITE, DEFAULT, NONE, null, null, subAttributeSchemaList);
+        AttributeSchema attributeSchema2 =
+                SCIMAttributeSchema.createSCIMAttributeSchema(
+                        "urn:ietf:params:scim:schemas:core:2.0:User:emails",
+                        "emails", COMPLEX,
+                        true, "", false, false,
+                        READ_WRITE, DEFAULT, NONE, null, null, subAttributeSchemaList1);
+        SCIMResourceTypeSchema scimResourceTypeSchema =
+                SCIMResourceTypeSchema.createSCIMResourceSchema(schemasList, attributeSchema1, attributeSchema2);
+
+        return new Object[][]{
+
+                {patchOperation, oldResource, copyOfOldResource, scimResourceTypeSchema, "BAD_REQUEST"},
+                {patchOperation1, oldResource, copyOfOldResource, scimResourceTypeSchema, "NOT_IMPLEMENTED"},
+                {patchOperation2, oldResource, copyOfOldResource, scimResourceTypeSchema, "SUCCESS"},
+                {patchOperation3, oldResource, copyOfOldResource, scimResourceTypeSchema, "SUCCESS"},
+                {patchOperation4, oldResource, copyOfOldResource, scimResourceTypeSchema, "NOT_IMPLEMENTED"}
+        };
+    }
+
+    @Test(dataProvider = "dataForPatchRemove")
+    public void testDoPatchRemove(PatchOperation operation, AbstractSCIMObject oldResource,
+                                  AbstractSCIMObject copyOfOldResource, SCIMResourceTypeSchema schema, String expect)
+            throws CharonException, BadRequestException, NotImplementedException, InternalErrorException {
+
+        String result = "";
```

Comment:
There is no need to initialize the variable here since it gets a value either in try block or catch block.Just declaration is enough
```suggestion
        String result;
```

Reply:
addressed with e042f4aba02cbad5a048d7e7ad9302231fecf217